### DISCUSSION TITLE
problem with get property from container in new version 3.0.0

### DISCUSSION THREAD

**Discussion by meysamhadeli** [testcontainers/testcontainers-dotnet#831] [Q&A]
In previous version, I can get property for container. For example, I get property `username`, `password`, `port` from container property. But in new version I don't access those properties. I can only access to `GetConnectionString()` method.
This is my code that work in previous version:
```csharp
    private void AddCustomAppSettings(IConfigurationBuilder configuration)
    {
        configuration.AddInMemoryCollection(new KeyValuePair<string, string>[]
        {
            new("RabbitMqOptions:HostName", RabbitMqTestContainer.Hostname),
            new("RabbitMqOptions:UserName", RabbitMqTestContainer.Username),
            new("RabbitMqOptions:Password", RabbitMqTestContainer.Password),
            new("RabbitMqOptions:Port", RabbitMqTestContainer.Port.ToString()),
        });
    }
```
Can we access to container properties in new version?

**Comment by HofmeisterAn** [comment]
> Can we access to container properties in new version?

Since you need to pass the credentials to the builder anyway, does it make sense to delegate the value around? I am not sure if it adds much value, but I'm happy to discuss it. Please keep in mind that if you stick to non-randomly generated credentials, you will be able to use the reusable feature in the future. This won't work with randomly generated credentials.

**Reply by meysamhadeli** [reply]
Can I `create` an `example` and send it to you, and we can discuss that? We have something like this in old version. But in new on we don't have very flexibility. It's a suggestion and I think it might be useful for others.

**Reply by HofmeisterAn** [reply]
Some pseudo code or code snippet that explains more about the context should be enough. We do not need an entire example. Personally, I am okay with passing values around, but I am uncertain whether it's essential to do so.
